### PR TITLE
chore: hook pricing table to billing service plans

### PR DESCRIPTION
**Pull request by raquelmsmith**
**⚠️ https://github.com/PostHog/billing/pull/78 must be merged first! ⚠️** 

## Problem

We want the pricing table to read from the billing service, which will soon be the source of truth for a customer's plan, pricing, and feature availability.

## Changes

Uses the new billing service `plans` API to generate available plans and populate the billing table. 

Still to do:
- [x] merge https://github.com/PostHog/billing/pull/78 first!!
- [x] dynamically set which plans to show
- [x] make sure the upgrade buttons actually work

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by raquelmsmith** [comment]
@kappa90 before merging we need to remember to:

- [x] Make sure we have a Stripe product with `product_type: base` and an associated price with `plan: enterprise` before merging this. Can I just use the "Copy to Live Mode" button on the [product](https://dashboard.stripe.com/test/products/prod_NEJOQbYQpnNHmy) that we made in the stripe dev env today?
    - image
    - We also need to make sure to use that new price ID in the yaml on the billing service.
    - I'm happy to do this but wanted to double check with you before creating the new product in stripe.
 - [ ] Deploy billing service after adding the correct price to the yaml

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:50]: This is too hardcoded for the scope of what we're doing :D

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:104]: Same as above, why is this hardcoded?

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:196]: We can retrieve the unit from Stripe, for the future

File: frontend/src/types.ts
Comment (kappa90) [inline frontend/src/types.ts:1068]: Why do we need these upper and lower case?

Comment (raquelmsmith) [reply frontend/src/types.ts:1068]: Because we are reusing existing types and extending them to work with the new plans API. The billing service returns products with uppercased product types, but the plans API returns them lowercase. Could update the billing service to always return lowercase, but I just made it work for now.

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (raquelmsmith) [reply frontend/src/scenes/billing/v2/PlanTable.tsx:50]: Oops, will update.

File: frontend/src/lib/constants.tsx
Comment (kappa90) [inline frontend/src/lib/constants.tsx:173]: Not sure if we need to implement this now, but maybe on a subsequent PR where we use the `available_features` list from the billing service to check for feature access. I'm worried this could break existing paygates.

**Review by kappa90 — CHANGES_REQUESTED** [review]
Left some comments!

File: frontend/src/scenes/instance/Licenses/licenseLogic.ts
Comment (raquelmsmith) [reply frontend/src/scenes/instance/Licenses/licenseLogic.ts]: This was because I was following type errors around after I extended the list of AvailableFeatures. But I realized I was dumb and could just use a `Partial<>` for the minimum features and avoid needing to do this.

File: frontend/src/types.ts
Comment (raquelmsmith) [reply frontend/src/types.ts]: I'm actually not seeing the uppercase ones any more so I just changed to all lower

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (raquelmsmith) [reply frontend/src/scenes/billing/v2/PlanTable.tsx]: Done.

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:134]: is_free again?

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:138]: same here

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:140]: same same

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:145]: same same same

File: frontend/src/scenes/billing/v2/PlanTable.tsx
Comment (kappa90) [inline frontend/src/scenes/billing/v2/PlanTable.tsx:266]: I guess this needs to be removed

**Review by kappa90 — APPROVED** [review]
Approving so you can merge, but there's a few more things I found

**Comment by raquelmsmith** [comment]
k all comments should be addressed. It should be noted that this whole plan table assumes that anyone who is seeing it is on the free plan. Eg. the upgrade buttons say 'current plan' on the free plan regardless. If we ever want to show it to people who are not on the free plan then we will need to refactor a bit. But that's out of scope here, we'll get this out tomorrow.

**Pull request merged**
